### PR TITLE
Make sure to return 404 when no events are found

### PR DESCRIPTION
**Pull request by t-persson**
### Applicable Issues

Fixes: #34 

### Description of the Change

There will now be a 404 if we do not find any events on a query. I also cleaned up so that we never return JSON payloads on errors.

### Alternate Designs

I decided to actually send a response payload in the form of the HTTP error status message. This, in my opinion, makes it more clear than letting the browser scream "PAGE NOT FOUND!" when you don't find any events.

### Benefits

We will follow the API specification and the risk of returning internal information to the users is reduced.

### Possible Drawbacks

None that I can think of.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@axis.com>

### CODE REVIEW

File: internal/responses/responses.go [hunk 1]
```diff
@@ -29,7 +29,8 @@ func RespondWithJSON(w http.ResponseWriter, code int, payload interface{}) {
 	_, _ = w.Write(response)
 }
 
-// RespondWithError writes a JSON response with an error message and status code to the HTTP ResponseWriter.
+// RespondWithError writes a response with an error message and status code to the HTTP ResponseWriter.
 func RespondWithError(w http.ResponseWriter, code int, message string) {
-	RespondWithJSON(w, code, map[string]string{"error": message})
+	w.WriteHeader(code)
+	_, _ = w.Write([]byte(message))
```
Comment (magnusbaeck) [inline internal/responses/responses.go:35]: Who or what is setting the content type in this response? The documentation isn't super clear on this.

**Review by magnusbaeck — COMMENTED** [review]
Just a question, LGTM otherwise.

Note to self: Make sure existing clients are okay with this behavioral change before deploying internally.

File: internal/responses/responses.go (hunk 1 above)
Comment (t-persson) [reply internal/responses/responses.go:35]: This was a rabbit-hole :) I'm pretty sure it is written here (same version of Golang as in the Dockerfile):
https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/net/http/server.go;l=1386

But it scares me a bit that we are letting Golang detect the content type so I will update the code to be more explicit :)

Comment (magnusbaeck) [reply internal/responses/responses.go]: Ah, yes, that's where the autodetection kicks in. Being explicit about this SGTM.

File: internal/responses/responses.go [hunk 2]
```diff
@@ -29,7 +29,9 @@ func RespondWithJSON(w http.ResponseWriter, code int, payload interface{}) {
 	_, _ = w.Write(response)
 }
 
-// RespondWithError writes a JSON response with an error message and status code to the HTTP ResponseWriter.
+// RespondWithError writes a response with an error message and status code to the HTTP ResponseWriter.
 func RespondWithError(w http.ResponseWriter, code int, message string) {
-	RespondWithJSON(w, code, map[string]string{"error": message})
+	w.WriteHeader(code)
+	w.Header().Set("Content-Type", "text/plan; charset=utf-8")
```
Comment (magnusbaeck) [inline internal/responses/responses.go:35]:
```suggestion
	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
```

**Review by magnusbaeck — CHANGES_REQUESTED** [review]
Almost :-)

**Review by magnusbaeck — APPROVED** [review]

**Pull request merged**